jax/_src/core.py

```diff
@@ -1935,7 +1935,9 @@ def symbolic_equal_dim(d1: DimSize, d2: DimSize) -> bool:
   return handler.symbolic_equal(*ds)
 
 def symbolic_equal_one_of_dim(d1: DimSize, dlist: Sequence[DimSize]) -> bool:
-  if any(d1 is d for d in dlist): return True  # identical always implies equal
+  # TODO(reviewer): Is there a reason that `same_referent` wasn't
+  # included here before?
```

Comment:
Not that I can think of... we're just adding it places on demand for dynamic shape / pile support. Seems good!

Reply:
Keeping.